File: Week6/sentimental-credit/credit.py
num = int(input("Number: "))

digit, counter, sum = 0, 0, 0
while (num > 0):
    counter += 1
    digitbuf = digit
    digit = num % 10
    if counter % 2 == 0:
        sum += (2 * digit) % 10 + (2 * digit) // 10
    else:
        sum += digit
    num = num // 10

cardtype = "INVALID"
if sum % 10 == 0:
    if digit == 3 and counter == 15 and (digitbuf == 4 or digitbuf == 7):
        cardtype = "AMEX"
    elif digit == 4 and (counter == 13 or counter == 16):
        cardtype = "VISA"
    elif digit == 5 and counter == 16 and (digitbuf > 0 and digitbuf < 6):
        cardtype = "MASTERCARD"
print(cardtype)

File: Week1/credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    //get input
    long num = get_long("Number: ");

    //calc
    int pointer = 0, digit = 0, digitbuf = 0, sum = 0;
    while (num > 0)
    {
        digitbuf = digit;
        digit = num % 10;
        pointer++;
        if (pointer % 2 == 0)
        {
            sum += (digit * 2) % 10 + (digit * 2) / 10;
        }
        else
        {
            sum += digit;
        }
        num = num / 10;
    }

    //output
    string cardtype = "INVALID";
    if (sum % 10 == 0)
    {
        switch (digit)
        {
            case 3:
                if (pointer == 15 && (digitbuf == 4 || digitbuf == 7))
                {
                    cardtype = "AMEX";
                }
                break;

            case 4:
                if (pointer == 13 || pointer == 16)
                {
                    cardtype = "VISA";
                }
                break;

            case 5:
                if (pointer == 16 && (digitbuf > 0 && digitbuf < 6))
                {
                    cardtype = "MASTERCARD";
                }
                break;

            default:
                break;
        }
    }
    printf("%s\n", cardtype);
}